src/Symfony/Component/Routing/Loader/XmlFileLoader.php

```diff
@@ -107,16 +107,20 @@ public function supports($resource, $type = null)
      */
     protected function parseRoute(RouteCollection $collection, \DOMElement $node, $path)
     {
-        if ('' === ($id = $node->getAttribute('id')) || !$node->hasAttribute('path')) {
-            throw new \InvalidArgumentException(sprintf('The <route> element in file "%s" must have an "id" and a "path" attribute.', $path));
+        if ('' === ($id = $node->getAttribute('id')) || (!$node->hasAttribute('path') && 'true' !== $node->getAttribute('empty-path'))) {
+            throw new \InvalidArgumentException(sprintf('The <route> element in file "%s" must have an "id" and a "path" (or "empty-path") attribute.', $path));
+        }
+
+        if ($node->hasAttribute('path') && 'true' === $node->getAttribute('empty-path')) {
```

Comment:
Since empty-path=false is useless, I would forbid specifing both `($node->hasAttribute('path') && $node->hasAttribute('empty-path'))`

Reply:
I think that's already covered by the previous check

Reply:
The solution with empty-path or path attribute is very nice